spec/decorators/course_decorator_spec.rb

```diff
@@ -197,4 +197,202 @@
+        it "returns the course name" do
+          expect(decorated_course.subject_name).to eq("Mathematics")
+        end
+      end
+
+      context "course has one subject" do
+        let(:subject) { build :subject, subject_name: "Computer Science" }
+        let(:course) { build :course, subjects: [subject] }
+
+        it "return the subject name" do
+          expect(decorated_course.subject_name).to eq("Computer Science")
+        end
+      end
+    end
+
+    describe "#requirements" do
+      let(:subject) { decorated_course.requirements }
+
+      context "Course with mathematics as a subject" do
+        let(:mathematics) { build :subject, :mathematics, subject_name: "Primary with Mathematics" }
+        let(:english) { build :subject, :english }
+        let(:subjects) { [mathematics, english] }
+
+        expected_requirements = [
+          "a degree of 2:2 or above in any subject",
+          "at least grade B in maths A-level (or an equivalent)",
+        ]
+
+        it { is_expected.to eq(expected_requirements) }
+      end
+
+      context "Course without mathematics as a subject" do
+        let(:english) { build :subject, :english }
+        let(:subjects) { [biology, english] }
+
+        expected_requirements = [
+          "a degree of 2:2 or above in any subject",
+        ]
+
+        it { is_expected.to eq(expected_requirements) }
+      end
+    end
+
+    describe "#bursary_first_line_ending" do
+      let(:subject) { decorated_course.bursary_first_line_ending }
+
+      context "More than one requirement" do
+        let(:mathematics) { build :subject, :mathematics, subject_name: "Primary with Mathematics" }
+        let(:english) { build :subject, :english }
+        let(:subjects) { [mathematics, english] }
+
+        expected_line_ending = ":"
+
+        it { is_expected.to eq(expected_line_ending) }
+      end
+
+      context "Course without mathematics as a subject" do
+        let(:english) { build :subject, :english }
+        let(:subjects) { [biology, english] }
+
+        expected_line_ending = "a degree of 2:2 or above in any subject."
+
+        it { is_expected.to eq(expected_line_ending) }
+      end
+    end
+
+
+    describe "#bursary_only" do
+      let(:subject) { decorated_course }
+
+      context "course only has bursary financial incentives" do
+        let(:mathematics) { build :subject, bursary_amount: "2000" }
+        let(:english) { build :subject, bursary_amount: "4000" }
+        let(:subjects) { [mathematics, english] }
+
+        it { is_expected.to be_bursary_only }
+      end
+
+      context "course has other financial incentives apart from bursaries" do
+        let(:mathematics) { build :subject, bursary_amount: "2000" }
+        let(:english) { build :subject, scholarship: "4000" }
+        let(:subjects) { [mathematics, english] }
+
+        it { is_expected.to_not be_bursary_only }
+      end
+    end
+
+    describe "Bursaries" do
```

Comment:
not sure about this?

maybe is a context about all `bursary` like you did with `scholarships`